pkg/component/gardeneradmissioncontroller/secrets.go

```diff
@@ -0,0 +1,40 @@
+// Copyright 2023 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package gardeneradmissioncontroller
+
+import (
+	"context"
+
+	corev1 "k8s.io/api/core/v1"
+
+	operatorv1alpha1 "github.com/gardener/gardener/pkg/apis/operator/v1alpha1"
+	kubernetesutils "github.com/gardener/gardener/pkg/utils/kubernetes"
+	secretsutils "github.com/gardener/gardener/pkg/utils/secrets"
+	secretsmanager "github.com/gardener/gardener/pkg/utils/secrets/manager"
+)
+
+func (a admissioncontroller) reconcileSecretServer(ctx context.Context) (*corev1.Secret, error) {
```

Comment:
For all other components we have pointer receivers.

```suggestion
func (a *admissioncontroller) reconcileSecretServer(ctx context.Context) (*corev1.Secret, error) {
```